crates/pop-cli/tests/contract.rs

```diff
@@ -43,8 +43,12 @@ async fn contract_lifecycle() -> Result<()> {
 	assert!(temp_dir.join("test_contract/target/ink/test_contract.json").exists());
 
 	// Run the contracts node
-	let cache = temp_dir.join("cache");
-	let process = run_contracts_node(cache, None).await?;
+	// temp_dir gets dropped prematurely, so manually create a tmp directory.
+	let cache = dirs::cache_dir().expect("cache_dir failed").join("pop_tmp_contract_lifecycle");
```

Comment:
It's because the temp_dir handle gets moved in the subsequent functioned calls, so, although the original handle still exists (when using clone), the clones get dropped which result in the directory getting prematurely cleaned.

Passing references introduces unnecessary complexity in terms of manually specifying lifetimes on the `Binary` struct. So, I felt manually creating a temp_dir is the least overhead for a test.

Regarding the leftover tmp, in-practice a failing test is usually re-run until it is fixed. I will continue looking for a better way.

Reply:
Does temp_dir.path().join() not work?